src/Nethermind/Nethermind.JsonRpc/Modules/Subscribe/SubscriptionFactory.cs

```diff
@@ -48,25 +51,29 @@ public SubscriptionFactory(ILogManager? logManager,
         filterStore = filterStore ?? throw new ArgumentNullException(nameof(filterStore));
         ethSyncingInfo = ethSyncingInfo ?? throw new ArgumentNullException(nameof(ethSyncingInfo));
         specProvider = specProvider ?? throw new ArgumentNullException(nameof(specProvider));
+        _peerPool = peerPool ?? throw new ArgumentNullException(nameof(peerPool));
 
         _subscriptionConstructors = new ConcurrentDictionary<string, CustomSubscriptionType>
         {
 
             //Register the standard subscription types in the dictionary.
-            [SubscriptionType.NewHeads] = CreateSubscriptionType<TransactionsOption?>((jsonRpcDuplexClient, args) =>
+            [SubscriptionType.EthSubscription.NewHeads] = CreateSubscriptionType<TransactionsOption?>((jsonRpcDuplexClient, args) =>
                 new NewHeadSubscription(jsonRpcDuplexClient, blockTree, logManager, specProvider, args)),
 
-            [SubscriptionType.Logs] = CreateSubscriptionType<Filter?>((jsonRpcDuplexClient, filter) =>
+            [SubscriptionType.EthSubscription.Logs] = CreateSubscriptionType<Filter?>((jsonRpcDuplexClient, filter) =>
                 new LogsSubscription(jsonRpcDuplexClient, receiptCanonicalityMonitor, filterStore, blockTree, logManager, filter)),
 
-            [SubscriptionType.NewPendingTransactions] = CreateSubscriptionType<TransactionsOption?>((jsonRpcDuplexClient, args) =>
+            [SubscriptionType.EthSubscription.NewPendingTransactions] = CreateSubscriptionType<TransactionsOption?>((jsonRpcDuplexClient, args) =>
                 new NewPendingTransactionsSubscription(jsonRpcDuplexClient, txPool, specProvider, logManager, args)),
 
-            [SubscriptionType.DroppedPendingTransactions] = CreateSubscriptionType(jsonRpcDuplexClient =>
+            [SubscriptionType.EthSubscription.DroppedPendingTransactions] = CreateSubscriptionType(jsonRpcDuplexClient =>
                 new DroppedPendingTransactionsSubscription(jsonRpcDuplexClient, txPool, logManager)),
 
-            [SubscriptionType.Syncing] = CreateSubscriptionType(jsonRpcDuplexClient =>
-                new SyncingSubscription(jsonRpcDuplexClient, blockTree, ethSyncingInfo, logManager))
+            [SubscriptionType.EthSubscription.Syncing] = CreateSubscriptionType(jsonRpcDuplexClient =>
+                new SyncingSubscription(jsonRpcDuplexClient, blockTree, ethSyncingInfo, logManager)),
+
+            [SubscriptionType.AdminSubscription.PeerEvents] = CreateSubscriptionType((jsonRpcDuplexClient) =>
+                new PeerEventsSubscription(jsonRpcDuplexClient, logManager, _peerPool)),
```

Comment:
I mean this kind of registrations could be done when setting up `AdminModule` and previous ones when setting up `SubscribeModule`. Would make this class bit simpler to construct and use. We could use extension methods for those registrations.